src/main/java/com/synopsys/integration/alert/web/security/authentication/saml/AlertSAMLMetadataGenerator.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.web.security.authentication.saml;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.saml.metadata.MetadataGenerator;
+
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.exception.AlertLDAPConfigurationException;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.component.settings.SettingsDescriptor;
+
+public class AlertSAMLMetadataGenerator extends MetadataGenerator {
+    private static final Logger logger = LoggerFactory.getLogger(AlertSAMLMetadataGenerator.class);
+    private final SAMLContext samlContext;
+
+    public AlertSAMLMetadataGenerator(final SAMLContext samlContext) {
+        this.samlContext = samlContext;
+    }
+
+    public String getEntityId() {
+        try {
+            final ConfigurationModel currentConfiguration = samlContext.getCurrentConfiguration();
+            return samlContext.getFieldValueOrEmpty(currentConfiguration, SettingsDescriptor.KEY_SAML_ENTITY_ID);
+        } catch (final AlertDatabaseConstraintException | AlertLDAPConfigurationException e) {
+            logger.error("Could not get the SAML entity ID.", e);
+        }
+        return "";
+    }
+
+    public String getEntityBaseURL() {
+        try {
+            final ConfigurationModel currentConfiguration = samlContext.getCurrentConfiguration();
+            return samlContext.getFieldValueOrEmpty(currentConfiguration, SettingsDescriptor.KEY_SAML_ENTITY_BASE_URL);
+        } catch (final AlertDatabaseConstraintException | AlertLDAPConfigurationException e) {
+            logger.error("Could not get the SAML entity base URL.", e);
+        }
+        return "";
+    }
```

Comment:
These two methods can probably use another helper method to cut out redundant code.